src/ophyd_async/core/detector.py

```diff
@@ -68,7 +68,7 @@ async def open(self, multiplier: int = 1) -> Dict[str, Descriptor]:
         """
 
     @abstractmethod
-    async def wait_for_index(self, index: int) -> None:
+    async def wait_for_index(self, index: int, timeout: Optional[float] = None) -> None:
```

Comment:
I suggest we make this `DEFAULT_TIMEOUT` so people have to opt into a "wait forever" behaviour
```suggestion
    async def wait_for_index(self, index: int, timeout: Optional[float] = DEFAULT_TIMEOUT) -> None:
```